spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/DatagramPacketMessageMapper.java

```diff
@@ -142,7 +145,23 @@ public DatagramPacket fromMessage(Message<?> message) throws Exception {
 			buffer.put(bytes);
 			bytes = buffer.array();
 		}
-		return new DatagramPacket(bytes, bytes.length);
+		return getDatagramFromHeader(message, bytes, bytes.length);
+	}
+
+	private DatagramPacket getDatagramFromHeader(Message<?> message, byte[] bytes,
+												 int length) throws UnknownHostException {
+		InetAddress inetAddress = null;
+		if (message.getHeaders().get(IpHeaders.IP_ADDRESS, String.class) != null) {
+			inetAddress = InetAddress.getByName(message.getHeaders()
+					.get(IpHeaders.IP_ADDRESS, String.class));
```

Comment:
The sending adapters just uses the result of `socket-expression`, so, I wonder why we need the:
```
* @param   address  the destination address.
* @param   port     the destination port number.
```
In the `DatagramPacket` at all?

Sorry, I might don't understand UDP properly and we really should use both options (`destinationAddress` and `socketExpression`)...